### DISCUSSION TITLE
Supabase access token and google Auth access token are different ?

### DISCUSSION THREAD

**Discussion by kavya-gaur** [supabase/supabase#27568] [Questions]
Hello fellas,

I am experiencing an issue where I am using supabase auth for handling the OAuth flow. My goal is to get email messages of currently logged in user. This is not possible with the person's access token. Now I figured out that supabase session object provides us the access token. I used it but it didn't work. Since supabase access token and google one is totally different. I am wondering if there is another alternative to do that without explicitly handing the OAuth flow myself. Looking forward to your responses.

**Comment by GaryAustin1** [comment]
You get the provider tokens when you first signup/in with an Oauth provider in the session object.  It is not saved though and will go away on refresh of the session/jwt.  https://supabase.com/docs/guides/auth/social-login#provider-tokens

**Reply by kavya-gaur** [reply]
Hey. Thanks for your reply. Yes I just read the documentation. However, when i use provider_token which i receive the very first time, it says its expired. However I cannot refresh the token too since I am not getting any provider_refresh_token from supabase. Can you help me out if there are any additional steps to get google refresh token ? I can manage to refresh it explicitly but would need refresh token.

**Reply by GaryAustin1** [reply]
Sorry, above is what I know about the process.  There are a few hits here on google refresh token and this:  https://github.com/supabase/auth/issues/1387  So not real clear to me the process.  I was mainly pointing out access_token is not from OAuth and different from provider_token.
